Islandora/islandora
# Version files

**GitHub Issue**: (https://github.com/Islandora/documentation/issues/740)

# What does this Pull Request do?
Connects the replacement of a Fedora flysystem file in Drupal to the versioning of an existing resource in Fedora.

# What's new?
* requires a third party module now - file_replace
* extensions to the flysystem adapter

# How should this be tested?

* Create an islandora object with an attached media and file (being stored in fedora)
* install the file_replace module  (`composer require drupal/file_replace` and `drush pm-enable file_replace` (or use this PR: https://github.com/Islandora-Devops/islandora-playbook/pull/184)
* pull down this PR 
* go to the file replace page for your file like `/admin/content/files/replace/{{fid}}`
* replace the file
* get your fedora url for that file - ie. `http://localhost:8080/fcrepo/rest/2020-08/Barton_CHANS_2010.pdf`
* hit the fcr:versions endpoint for - you should see a new version (should have an ldp:contains at the end that ends with the date stamp for when you made the version) ie. `curl -v http://localhost:8080/fcrepo/rest/2020-08/Barton_CHANS_2010.pdf/fcr:versions`

# Additional Notes:
I'll need to package up a PR to islandora-playbook to add this module if people are ok with it

# Interested parties
@Islandora/8-x-committers
